src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java

```diff
@@ -0,0 +1,227 @@
+
+    @Description("The Bigtable table id to export.")
+    ValueProvider<String> getBigtableTableId();
+
+    @SuppressWarnings("unused")
+    void setBigtableTableId(ValueProvider<String> tableId);
+
+    @Description("The output location to write to (e.g. gs://mybucket/somefolder/)")
+    ValueProvider<String> getOutputDirectory();
+
+    @SuppressWarnings("unused")
+    void setOutputDirectory(ValueProvider<String> outputDirectory);
+
+    @Description(
+        "The prefix for each exported file in outputDirectory. Default file prefix is set to \"output\".")
+    @Default.String("output")
+    ValueProvider<String> getFilenamePrefix();
+
+    @SuppressWarnings("unused")
+    void setFilenamePrefix(ValueProvider<String> filenamePrefix);
+
+    @Description(
+        "The maximum number of output shards produced when writing. Default number is runner defined. ")
+    @Default.Integer(0)
+    ValueProvider<Integer> getNumShards();
+
+    @SuppressWarnings("unused")
+    void setNumShards(ValueProvider<Integer> numShards);
+  }
+
+  /**
+   * Main entry point for pipeline execution.
+   *
+   * @param args Command line arguments to the pipeline.
+   */
+  public static void main(String[] args) {
+    Options options = PipelineOptionsFactory.fromArgs(args).withValidation().as(Options.class);
+
+    PipelineResult result = run(options);
+
+    // Wait for pipeline to finish only if it is not constructing a template.
+    if (options.as(DataflowPipelineOptions.class).getTemplateLocation() == null) {
+      result.waitUntilFinish();
+    }
+  }
+
+  /**
+   * Runs a pipeline to export data from a Cloud Bigtable table to Parquet file(s) in GCS.
+   *
+   * @param options arguments to the pipeline
+   */
+  public static PipelineResult run(Options options) {
+    Pipeline pipeline = Pipeline.create(options);
+    BigtableIO.Read read =
+        BigtableIO.read()
+            .withProjectId(options.getBigtableProjectId())
+            .withInstanceId(options.getBigtableInstanceId())
+            .withTableId(options.getBigtableTableId());
+
+    // Do not validate input fields if it is running as a template.
+    if (options.as(DataflowPipelineOptions.class).getTemplateLocation() != null) {
+      read = read.withoutValidation();
+    }
+
+    /**
+     * Steps:
+     * 1) Read records from Bigtable.
+     * 2) Convert a Bigtable Row to a GenericRecord.
+     * 3) Write GenericRecord(s) to GCS in parquet format.
+     */
+    pipeline
+        .apply("Read from Bigtable", read)
+        .apply("Transform to Parquet", MapElements.via(new BigtableToParquetFn()))
+        .setCoder(AvroCoder.of(GenericRecord.class, BigtableRow.getClassSchema()))
+        .apply(
+            "Write to Parquet in GCS",
+            FileIO.<GenericRecord>write()
+                .via(ParquetIO.sink(BigtableRow.getClassSchema()))
+                .to(options.getOutputDirectory())
+                .withPrefix(options.getFilenamePrefix())
```

Comment:
As a test, can you create a template and then run the template by passing a specific --filenamePrefix=my-output

Reply:
@sabhyankar 
Create a template as in? 
Cause this is a legacy template.

Reply:
Yes - Create a legacy template and then call gcloud dataflow jobs run with that template and pass fileNamePrefix=my-output as a parameter.

Reply:
Okay got it

Reply:
@sabhyankar 
It works fine.